File: week2/esadmsk_w2/daily-temperatures.py
class Solution(object):
    def dailyTemperatures(self, temperatures):
        stack = []
        result = [0] * len(temperatures)

        for i in range(len(temperatures)):
            while stack and temperatures[i] > temperatures[stack[-1]]:
                index = stack.pop()
                result[index] = i - index
            stack.append(i)

        return result

File: week2/omerSefaYilmaz/daily-temperatures.js
/**
 * @param {number[]} temperatures
 * @return {number[]}
 */
var dailyTemperatures = function (temperatures) {
    let stack = [];
    let result = new Array(temperatures.length).fill(0);
  
    for (let i = temperatures.length - 1; i >= 0; i--) {
      while (stack.length > 0 && temperatures[i] >= stack[stack.length - 1].key) {
        stack.pop();
      }
  
      if (stack.length > 0) {// found warmer day
        result[i] = stack[stack.length - 1].value - i;
      }
  
      stack.push({ key: temperatures[i], value: i });
    }
  
    return result;
  };
  